source/Octopus.Cli/Commands/InstallAutoCompleteCommand.cs

```diff
@@ -0,0 +1,181 @@
+                {
+                    commandOutputProvider.Information("Looks like this is already installed. Bailing out.");
+                    return;
+                }
+
+                commandOutputProvider.Information($"Updating profile at {profilePath}");
+                tempOutput.AppendLine(profileText);
+            }
+            else
+            {
+                commandOutputProvider.Information($"Creating profile at {profilePath}");
+            }
+
+            tempOutput.AppendLine(UserProfileHelper.AllShellsPrefix);
+            tempOutput.AppendLine(scriptToInject);
+            tempOutput.AppendLine(UserProfileHelper.AllShellsSuffix);
+
+            if (DryRun)
+            {
+                commandOutputProvider.Warning("Preview of script changes: ");
+                commandOutputProvider.Information(tempOutput.ToString());
+                commandOutputProvider.Warning("Preview of script changes finished. ");
+            }
+            else
+            {
+                fileSystem.OverwriteFile(profilePath, tempOutput.ToString());
+                commandOutputProvider.Warning("All Done! Please reload your shell or dot source your profile to get started! Use the <tab> key to autocomplete.");
+            }
+        }
+        
+        private void InstallForBash()
+        {
+            Install(
+                UserProfileHelper.BashProfile, 
+                UserProfileHelper.BashProfileScript);
+        }
+
+        private void InstallForZsh()
+        {
+            Install(
+                UserProfileHelper.ZshProfile, 
+                UserProfileHelper.ZshProfileScript);
+        }
+
+        private void InstallForPowershell()
+        {
+            var profilePath = UserProfileHelper.GetPwshProfileForOperatingSystem();
+            Install(profilePath, UserProfileHelper.PwshProfileScript);
+        }
+    }
+    public static class UserProfileHelper
+    {
+        public static readonly string EnvHome = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? "HOMEPATH" : "HOME";
+        public static readonly string HomeDirectory = System.Environment.GetEnvironmentVariable(EnvHome);
+        public static readonly string ZshProfile = $"{HomeDirectory}/.zshrc";
+        public const string ZshProfileScript = 
+@"_octo_zsh_complete()
+{
+    local completions=(""$(octo complete $words)"")
+    reply=( ""${(ps:\n:)completions}"" )
+}
+compctl -K _octo_zsh_complete octo";
+
+        public static readonly string PwshProfileLocationWindows = $"{HomeDirectory}\\Documents\\WindowsPowerShell\\Microsoft.PowerShell_profile.ps1";
+        public static readonly string PwshProfileLocationNix =  $"{HomeDirectory}/.config/powershell/Microsoft.PowerShell_profile.ps1";
+        public const string PwshProfileScript =
+@"Register-ArgumentCompleter -Native -CommandName octo -ScriptBlock {
```

Comment:
Does this need to say `octo.exe`???

Reply:
Not on linux, no. Probably another 'windows only' thing I need to cater to.